src/unix/solarish/mod.rs

```diff
@@ -994,6 +994,7 @@ pub const O_EXCL: ::c_int = 1024;
 pub const O_NOCTTY: ::c_int = 2048;
 pub const O_TRUNC: ::c_int = 512;
 pub const O_NOFOLLOW: ::c_int = 0x20000;
+pub const O_DIRECTORY: ::c_int = 0x1000000;
```

Comment:
It would be good to get confirmation from the Solaris folks that `O_DIRECTORY` exists there, and with the same value, before adding it at the top level like this (rather than the illumos-specific module)

Reply:
Seems to exist according to https://docs.oracle.com/cd/E26502_01/html/E29032/open-2.html. I'm not sure about the value though.

Reply:
I received confirmation that it's the same, so this should be fine.